### ISSUE TITLE
C# 7.0 Syntax Highlighting

### ISSUE THREAD

**Issue by DustinCampbell** [dotnet/csharp-tmLanguage#6]
_From @ivanz on August 28, 2016 15:46_

This issue track the support for syntax highlighting of C# 7.0 features: https://blogs.msdn.microsoft.com/dotnet/2016/08/24/whats-new-in-csharp-7-0/
- [x] Constructor and destructor expression bodied
- [ ] Ref returns and locals
- [ ] Literals: `_` separator and binary literals (`0b1010_1011_1100_1101_1110_1111`)
- [ ] Local functions
- [ ] Tuples
- [ ] Switch statements with patterns
- [ ] `is`-expressions with patterns
- [ ] Out variables

There is high potential for regressions here and we should totally implement #696 first

_Copied from original issue: OmniSharp/omnisharp-vscode#728_

**Comment by DustinCampbell** [comment]
I'm going to split this into multiple bugs so the features can be worked on in parallel.

**Comment by DustinCampbell** [comment]
Created #7, #8, #9, #10, #11, #12, #13, #14
